consensus/merge/src/main/java/org/hyperledger/besu/consensus/merge/MergeContext.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Hyperledger Besu Contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.consensus.merge;
+
+import org.hyperledger.besu.consensus.merge.blockcreation.PayloadIdentifier;
+import org.hyperledger.besu.ethereum.ConsensusContext;
+import org.hyperledger.besu.ethereum.core.Block;
+import org.hyperledger.besu.ethereum.core.BlockHeader;
+import org.hyperledger.besu.ethereum.core.Difficulty;
+import org.hyperledger.besu.ethereum.eth.sync.state.SyncState;
+
+import java.util.Optional;
+
+public interface MergeContext extends ConsensusContext {
+
+  MergeContext setSyncState(SyncState syncState);
+
+  MergeContext setTerminalTotalDifficulty(final Difficulty newTerminalTotalDifficulty);
+
+  void setIsPostMerge(final Difficulty totalDifficulty);
+
+  boolean isPostMerge();
+
+  boolean isSyncing();
+
+  void observeNewIsPostMergeState(final NewMergeStateCallback newMergeStateCallback);
```

Comment:
Tiny opinion: I woulda used a Future here, but then again I've been looking at a lot of Vert.x code :)

Reply:
I could see that in a general sense given the method name.  The only particular usage for this currently is as a callback to enable/disable mining though.  We don't benefit from asynchrony in that case, in fact I think it makes things harder to reason about.  Perhaps this method could be renamed to reflect its very specific one-time use case of disabling mining during the pre/post-merge transition.